node/flatpak_node_generator/providers/npm.py

```diff
@@ -369,9 +367,49 @@ async def generate_package(self, package: Package) -> None:
             # Get a unique name to use for the Git repository folder.
             name = f'{package.name}-{source.commit}'
             path = self.gen.data_root / 'git-packages' / name
-            self.git_sources[package.lockfile][path] = source
+            self.git_sources[package.lockfile][path] = (package.name, source)
             self.gen.add_git_source(source.url, source.commit, path)
 
+            url = urllib.parse.urlparse(source.url)
+            if url.netloc == 'git@github.com' or url.netloc == 'github.com':
+                url = url._replace(
+                    netloc='codeload.github.com', path=re.sub('.git$', '', url.path)
```

Comment:
should the `.` in `.git` here and below be escaped? might be useful to extract this into a separate `git_suffix = re.compile(r'\.git$')` and then use that for the substitutions below.

Reply:
Done, thanks.